packages/contracts-bedrock/src/periphery/monitoring/DisputeMonitorHelper.sol

```diff
@@ -0,0 +1,157 @@
+    /// @param _factory The factory of the dispute games.
+    /// @param _game The game to check.
+    /// @return isValid_ True if the game was created by the factory, false otherwise.
+    function isGameRegistered(IDisputeGameFactory _factory, IDisputeGame _game) public view returns (bool isValid_) {
+        // Grab the game and game data.
+        (GameType gameType, Claim rootClaim, bytes memory extraData) = _game.gameData();
+
+        // Grab the verified address of the game based on the game data.
+        (IDisputeGame _factoryRegisteredGame,) =
+            _factory.games({ _gameType: gameType, _rootClaim: rootClaim, _extraData: extraData });
+
+        // Return whether the game is factory registered.
+        isValid_ = address(_factoryRegisteredGame) == address(_game);
+    }
+
+    /// @notice Finds all unresolved games in a given time range.
+    /// @param _factory The factory of the dispute games.
+    /// @param _creationRangeStart Start of the range of game creation timestamps.
+    /// @param _creationRangeEnd End of the range of game creation timestamps.
+    /// @return unresolvedGames_ The array of unresolved games.
+    function getUnresolvedGames(
+        IDisputeGameFactory _factory,
+        uint256 _creationRangeStart,
+        uint256 _creationRangeEnd
+    )
+        public
+        view
+        returns (IDisputeGame[] memory unresolvedGames_)
+    {
+        // Check that the max
+        if (_creationRangeEnd < _creationRangeStart) {
+            revert DisputeMonitorHelper_InvalidSearchRange();
+        }
+
+        // If there are no games, return an empty array. In theory we could error here too but it's
+        // easier for offchain tooling if this case just returns empty. Either is fine, but this is
+        // likely to be a common standard case and it'd be nicer if it didn't error.
+        if (_factory.gameCount() == 0) {
+            return new IDisputeGame[](0);
+        }
+
+        // Try to find a suitable start and end index. If startIdx is type(uint256).max then we did
+        // not find any newer games and the creation range start must be after the timestamp of the
+        // latest game. Similarly, if endIdx is type(uint256).max then we did not find any older
+        // games and the creation range end must be before the timestamp of the earliest game. In
+        // either case, we can return an empty array.
+        uint256 startIdx = search(_factory, _creationRangeStart, SearchDirection.NEWER_THAN_OR_EQ);
+        uint256 endIdx = search(_factory, _creationRangeEnd, SearchDirection.OLDER_THAN_OR_EQ);
+        if (startIdx == type(uint256).max || endIdx == type(uint256).max) {
+            return new IDisputeGame[](0);
+        }
+
+        // Additionally, if the end index is less than the start index, then the range is between
+        // two dispute games. We return an empty array in this case.
+        if (endIdx < startIdx) {
```

Comment:
combining with the above if statement may create more efficient branching?

Reply:
Honestly efficiency not so important for this contract, was easier to explain the reason for the branches if they got separated